panel/main.cpp

```diff
@@ -48,65 +49,32 @@ void termSignalHandler(int)
         qApp->quit();
 }
 
-
-void printHelp()
-{
-    QTextStream out(stdout);
-    out << "Usage: lxqt-panel [options]" << endl;
-    out << endl;
-    out << "Start lxde-qt panel and its plugins" << endl;
-    out << endl;
-    out << "Options:" << endl;
-    out << "  -h, --help                    Show help about options" << endl;
-    out << "      --version                 Show version information" << endl;
-    out << "  -c, --configfile=CONFIGFILE   Use alternate configuration file" << endl;
-}
-
-
-void printVersion()
-{
-    QTextStream out(stdout);
-    out << "lxqt-panel " << LXQT_VERSION << endl;
-}
-
-
 int main(int argc, char *argv[])
 {
-    QString configFile;
-    for (int i=1; i < argc; ++i)
-    {
-        QString arg = QString::fromLocal8Bit(argv[i]);
 
-        if (arg == "--help" || arg == "-h")
-        {
-            printHelp();
-            return 0;
-        }
+    LxQtPanelApplication app(argc, argv);
+    QApplication::setApplicationName("lxqt-panel");
+    QApplication::setApplicationVersion(LXQT_VERSION);
 
-        if (arg == "--version")
-        {
-            printVersion();
-            return 0;
-        }
+    QCommandLineParser parser;
+    parser.setApplicationDescription("lxde-qt panel");
+    parser.addHelpOption();
+    parser.addVersionOption();
 
-        if (arg == "-c" || arg.startsWith("--conf"))
-        {
-            if (i+1 < argc)
-            {
-                configFile = QString::fromLocal8Bit(argv[i+1]);
-            }
-        }
-    }
+    QCommandLineOption configFileOption(QStringList()
+            << QStringLiteral("c") << QStringLiteral("config") << QStringLiteral("configfile"),
+            QCoreApplication::translate("main", "Use alternate configuration file."),
+            QCoreApplication::translate("main", "Configuration file"));
+    parser.addOption(configFileOption);
 
-    QScopedPointer<LxQtPanelApplication> app(new LxQtPanelApplication(argc, argv, configFile));
+    parser.process(app);
 
+    app.init(parser.value(configFileOption));
```

Comment:
The `QCoreApplication::arguments()` actually doesn't seem to work unless a `QCoreApplication` has been instantiated:

`QCoreApplication::arguments: Please instantiate the QApplication object first`

We could probably transform argv into a `QStringList` and use that, but then I'm not sure the `--version` or, the `--help` relying on the application's description would work well, I'm also unsure about the translations before the app's instantiation.

I don't like the `init` either, but using the main to build the application before starting it with `exec` makes sense, it's like an extended constructor. What would be nice would be to be able to construct an `LxQtPanelApplication` *from* an existing `QCoreApplication`.